alchemist/alchemist-cognitive-agents/src/main/kotlin/it/unibo/alchemist/characteristics/individual/HelpAttitude.kt

```diff
@@ -0,0 +1,53 @@
+package it.unibo.alchemist.characteristics.individual
+
+class HelpAttitude(age: Age, gender: Gender) : IndividualCharacteristic {
+
+    private val helperRules = rules[age to gender]
+
+    fun level(toHelpAge: Age, toHelpGender: Gender, sameGroup: Boolean): Double =
+            helperRules?.get(toHelpAge to toHelpGender)?.let {
+                if (sameGroup) it.first else it.second
+            } ?: 0.0
+
+    companion object {
```

Comment:
These look very much like a bunch of magic numbers better set in a property file... Don't they?

Reply:
I know but they are fixed, I took them from papers.

Reply:
It's fine, but stick them to a property file or resource bundle, or config file.

Reply:
ok, which is the preferable way to load them inside the class, which I suppose will be the same way I'll use to load the actually w, a, n prefixed parameters?

Reply:
Uhmm... I don't think there is a "preferred" way, besides avoiding custom formats and non-human-readable (e.g. XML). Java historically uses property files and resource bundles. Modern software configuration go in toml or yml, or json.
I would like to keep the simulator configuration as coherent as possible... see if there is any other sub-project and what it is using.
I am also thinking of a very nice idea... We could make a library that can load both TOML and YAML, and use that to load configurations in both formats. Could be used to allow writing simulations in TOML, which may make sense if you don't use variables.

Reply:
I'm a bit confused since I don't know what TOML is, anyway I'll try to look at if some other modules are importing some parameters or something similar

Reply:
Just [search for TOML](http://lmgtfy.com/?q=toml), it's a markup language similar to YAML, arguably simpler.

Reply:
Yes yes already done, just I don't get the main advantages of it over json for example

Reply:
Easier to read for humans

JSON is good for serialization, much less as configuration file. By the way, I believe that Java's property file syntax is a subset of TOML (basically a flat map).

Reply:
ok, I'll try creating it let's see if it's feasible

Reply:
Works if: there is a Java/Kt library that can easily transform TOML into a List or Map, as snakeyaml does for YAML. If so, it's just matter of using the appropriate tool when loading the file, the rest should be just dealing with maps.

Reply:
https://github.com/uchuhimo/konf

this one seems trustable

Reply:
WTF looks amazing

It may be the case to rewrite `YAMLLoader` using that library. Let's see how it works for the simpler stuff first, then we might want to go for the big fish.

Reply:
In order to access a file in resources, which is inside src/main, which path do I have to provide?

Reply:
done it was src/main/resources/parameters.toml since resources is not in the class path, but still having some problems

Reply:
Nope. Access is done via classpath, there is an internal thread-safe resource loader. **Never** point to source folders, they won't exist once deployed.
Simply access `parameters.toml`.

By the way, bad position and naming: shove it in `it.unibo.alchemist.somethingappropriategivenyourmodulepackages` and call with a name conveying information: parameters of what, exactly?